lib/types/query/order.ts

```diff
@@ -0,0 +1,68 @@
+import { FieldsType } from './util'
+import { EntryFields, EntrySys } from '../entry'
+import { AssetSys } from '../asset'
+import { ConditionalPick } from 'type-fest'
+import { TagSys } from '../tag'
+
+type SupportedTypes =
+  | EntryFields.Symbol
+  | EntryFields.Integer
+  | EntryFields.Number
+  | EntryFields.Date
+  | EntryFields.Boolean
+  | undefined
+
+type SupportedLinkTypes = EntryFields.AssetLink | EntryFields.EntryLink<any> | undefined
+
+export type OrderFilterPaths<Fields extends FieldsType, Prefix extends string> =
+  | `${Prefix}.${keyof ConditionalPick<Fields, SupportedTypes> & string}`
```

Comment:
Why do we need the `& string` here? Isn't `keyof FieldsType` always a string?

Reply:
In some cases it’s just string. Here we get `string | number` so we have to remove `number` to make TS happy.